src/System.Private.CoreLib/shared/System/Collections/Generic/List.cs

```diff
@@ -1117,13 +1118,13 @@ private bool MoveNextRare()
                 }
 
                 _index = _list._size + 1;
-                _current = default;
+                _current = default!;
                 return false;
             }
 
             public T Current => _current;
 
-            object IEnumerator.Current
+            object? IEnumerator.Current // TODO-NULLABLE-GENERIC: nullable only when T can be null
```

Comment:
While true that it could match, we probably don't need to comment cases like this.  This is an explicit interface implementation: there's nothing that shows up in the public API here that would allow for the consumer to receive fewer warnings if we could annotate this "correctly".

Reply:
will change since I don't feel strongly about it but IMO if T is say `int` you will currently get a warning and you should not IMO

Reply:
> if T is say int you will currently get a warning and you should not

Who will get a warning where?  The consumer of this?  What is the code the consumer would write that would result in a warning where that warning would be avoided if this were changed?

Reply:
isn't foreach using non-generic IEnumerable?

Reply:
Anything consuming this via the interface will see the interface's annotations, not these annotations.  This is an _explicit_ implementation of the interface; it is not public surface area.  The only way to call this is via the interface.